federation/pkg/federation-controller/namespace/namespace_controller_test.go

```diff
@@ -111,6 +111,12 @@ func TestNamespaceController(t *testing.T) {
 	assert.NotNil(t, createdNamespace)
 	assert.Equal(t, ns1.Name, createdNamespace.Name)
 
+	// Wait for the secret to appear in the informer store
+	err := WaitForStoreUpdate(
+		namespaceController.namespaceFederatedInformer.GetTargetStore(),
+		cluster1.Name, ns1.Name, 10*time.Second)
```

Comment:
Change 10s to `wait.ForeverTestTimeout`? It's 30s instead of 10, but it would be more consistent with other tests.